nati_bluefin_defconfig: Cherry-pick all nati_bluefin_defconfig related commits

DeborahOoi96 opened the pull request:
We need to update linux 6.6 kernel to support bluefin devices. Cherry-picking all nati_bluefin_defconfig related commits from bluefin/master/4.1 into nilrt/master/6.6. This file is needed in order for linux-bluefin kernel recipe to build correctly.

WI: [AB#2959863](https://dev.azure.com/ni/DevCentral/_workitems/edit/2959863)

Testing
Unable to build kernel until all necessary commits are cherry-picked. nati_bluefin_defconfig is only part of the changes needed to build.

bstreiff committed:
nati_bluefin_defconfig: add configuration

Bluefin-class devices are not deployable LabVIEW Real-Time targets,
and are sufficiently different from devices using nati_zynq_defconfig
to warrant their own configuration.

Signed-off-by: Brandon Streiff <brandon.streiff@ni.com>
Acked-by: Nathan Sullivan <nathan.sullivan@ni.com>
Natinst-ReviewBoard-ID: 158229
(cherry picked from commit e463c95779f89e2f19377da18cd5640c44c64a34)
Signed-off-by: deooi <deborah.ooi@ni.com>

bstreiff committed:
nati_bluefin_defconfig: disable RT scheduler

This is causing streaming performance issues, observed with a NI 9223
with one channel at max rate. Long term we'd like to have this on for
parity with NI Linux RT, but until we can investigate further, switch
back to PREEMPT_NONE to behave like the cDAQ-9188.

Signed-off-by: Brandon Streiff <brandon.streiff@ni.com>
Acked-by: Nathan Sullivan <nathan.sullivan@ni.com>
Natinst-ReviewBoard-ID: 158230
(cherry picked from commit 5299116dee79cf392b1b2b48c91637df693809a8)
Signed-off-by: deooi <deborah.ooi@ni.com>

bstreiff committed:
nati_bluefin_defconfig: add PHY LED triggers to configuration

Signed-off-by: Brandon Streiff <brandon.streiff@ni.com>
Acked-by: James Minor <james.minor@ni.com>
Natinst-ReviewBoard-ID: 191081
(cherry picked from commit 1fa2b96998a6cdbf9cd9cc6d6681b811b553b12b)
Signed-off-by: deooi <deborah.ooi@ni.com>

bstreiff committed:
nati_bluefin_defconfig: add si5351 driver to kernel config

Signed-off-by: Brandon Streiff <brandon.streiff@ni.com>
Acked-by: James Minor <james.minor@ni.com>
Natinst-ReviewBoard-ID: 191413
(cherry picked from commit 75bfb7e9fecce8691bd5715d7ce50b8543bdc02f)
Signed-off-by: deooi <deborah.ooi@ni.com>

ErikHons committed:
nati_bluefin_defconfig: enable 1588 and add ebtables/iptables support

Signed-off-by: Erik Hons <erik.hons@ni.com>
Acked-by: Brandon Streiff <brandon.streiff@ni.com>
Natinst-ReviewBoard-ID: 222750
(cherry picked from commit 63767b7ce6190cd1676a3310f3840dc6a1eaa592)
Signed-off-by: deooi <deborah.ooi@ni.com>

ErikHons committed:
nati_bluefin_defconfig: add netfilter mark match support

Allows ebtables to match dst MAC and iptables to match UDP port to
narrow PTP filter.

Signed-off-by: Erik Hons <erik.hons@ni.com>
Acked-by: Zach Brown <zach.brown@ni.com>
Acked-by: Brandon Streiff <brandon.streiff@ni.com>
Natinst-ReviewBoard-ID: 225068
(cherry picked from commit def7508ad78174177dc949c89709f5c1d39cbebf)
Signed-off-by: deooi <deborah.ooi@ni.com>

sqooi-ni committed:
nati_bluefin_defconfig: enable bonding driver

Enable network bonding to support link aggregation.

Signed-off-by: Song Qiang Ooi <song.qiang.ooi@ni.com>
(cherry picked from commit 3d890acbdc458ccc8f904b9b1521d9a2c15abf90)
Signed-off-by: deooi <deborah.ooi@ni.com>

chaitu236 reviewed (changes requested):
Can you do the cherry-picks with `-x` argument i.e., `git cherry-pick -x` so that the commit message contains the hash of the commit that was cherry-picked. Also, please sign the commits with your emerson mail id (i.e., git cherry-pick -x -s).

gratian commented:
Instead of cherry picking https://github.com/ni/linux/pull/186/commits/3fe3abdd652d5c1b7c35eecbd9da9b4fc6115c83 ("nati_bluefin_defconfig: round-trip through savedefconfig") can you re-generate it for 6.6 via savedefconfig (like that commit did for a previous version).

The steps would be:

1. Cherry pick all the other commits you have here (with the git options @chaitu236 mentioned above), minus commit https://github.com/ni/linux/pull/186/commits/3fe3abdd652d5c1b7c35eecbd9da9b4fc6115c83, and run the following commands in your Linux source tree
2. make nati_bluefin_defconfig
3. make savedefconfig
4. Copy the resulting 'defconfig' file over the nati_bluefin_defconfig
5. Create a new commit for nati_bluefin_defconfig with all the changes that happened due to savdefconfig. You can use a commit message similar to the one in https://github.com/ni/linux/pull/186/commits/3fe3abdd652d5c1b7c35eecbd9da9b4fc6115c83

Commit:
nati_bluefin_defconfig: round-trip through savedefconfig
Round-trip the configuration through savedefconfig so that it's ordered.

No functional changes to configuration otherwise.

Signed-off-by: deooi <deborah.ooi@ni.com>

gratian commented:
I've made a small tweak to the spacing in the commit message for "nati_bluefin_defconfig: round-trip through savedefconfig" to make the shortlog show up OK. (i.e. added an empty line after the subject).

PR pulled outside the GitHub workflow. Closing.
